control_toolbox/src/pid_ros.cpp

```diff
@@ -275,21 +349,45 @@ double PidROS::compute_command(double error, double error_dot, const rclcpp::Dur
 Pid::Gains PidROS::get_gains() { return pid_.get_gains(); }
 
 void PidROS::set_gains(double p, double i, double d, double i_max, double i_min, bool antiwindup)
+{
+  set_gains(p, i, d, i_max, i_min, 0.0, 0.0, 0.0, false, antiwindup, AntiwindupStrategy::NONE);
+}
+
+void PidROS::set_gains(
+  double p, double i, double d, double i_max, double i_min, double u_max, double u_min,
+  double trk_tc, bool saturation, bool antiwindup, AntiwindupStrategy antiwindup_strat)
 {
   if (i_min > i_max)
   {
     RCLCPP_ERROR(node_logging_->get_logger(), "received i_min > i_max, skip new gains");
   }
+  else if (u_min > u_max)
+  {
+    RCLCPP_ERROR(node_logging_->get_logger(), "received u_min > u_max, skip new gains");
+  }
   else
   {
+    if (antiwindup_strat == AntiwindupStrategy::NONE)
+    {
+      std::cout << "Old anti-windup technique is deprecated. "
+                   "This option will be removed by the ROS 2 Kilted Kaiju release."
```

Comment:
```suggestion
                   "This option will be removed after the ROS 2 Kilted Kaiju releases."
```
Should it be after? @christophfroehlich

Reply:
I would opt for backporting it as it comes with very nice features + easy transition

Reply:
it does not break API but ABI, and that's what I try to avoid :/